packages/firebase_auth/firebase_auth/ios/Classes/FLTFirebaseAuthPlugin.m

```diff
@@ -446,6 +453,57 @@ - (void)signInWithCredential:(id)arguments
                   }];
 }
 
+- (void)signInWithAuthProvider:(id)arguments
+          withMethodCallResult:(FLTFirebaseMethodCallResult *)result {
+#if TARGET_OS_OSX
+  NSLog(@"The Firebase Phone Authentication provider is not supported on the "
+        @"MacOS platform.");
+  result.success(nil);
+#else
+  FIRAuth *auth = [self getFIRAuthFromArguments:arguments];
+  self.authProvider = [FIROAuthProvider providerWithProviderID:arguments[@"signInProvider"]];
+
+  [self.authProvider
+      getCredentialWithUIDelegate:nil
+                       completion:^(FIRAuthCredential *_Nullable credential,
+                                    NSError *_Nullable error) {
+                         if (error) {
+                           result.error(kErrCodeInvalidCredential, kErrMsgInvalidCredential, nil,
+                                        nil);
+                           return;
+                         }
+                         if (credential) {
+                           [auth
+                               signInWithCredential:credential
+                                         completion:^(FIRAuthDataResult *authResult,
+                                                      NSError *error) {
+                                           if (error != nil) {
+                                             NSDictionary *userInfo = [error userInfo];
+                                             NSError *underlyingError =
+                                                 [userInfo objectForKey:NSUnderlyingErrorKey];
+
+                                             NSDictionary *firebaseDictionary =
+                                                 underlyingError.userInfo[@"FIRAuthErrorUserInfoDes"
+                                                                          @"erializedResponseKey"];
+
+                                             if (firebaseDictionary != nil &&
+                                                 firebaseDictionary[@"message"] != nil) {
+                                               // error from firebase-ios-sdk is
+                                               // buried in underlying error.
+                                               result.error(nil, firebaseDictionary[@"message"],
```

Comment:
no error code I guess?

Reply:
No error code here indeed (I've used the same way as the signInWithCredential function)